augur/io/metadata.py

```diff
@@ -362,3 +363,43 @@ def read_metadata_with_sequences(metadata, fasta, seq_id_column, seq_field='sequ
 
     if error_message:
         raise AugurError(f"Encountered the following error(s) when parsing metadata with sequences:{error_message}")
+
+
+def write_records_to_tsv(records, output_file):
+    """
+    Write each record from *records* as a single row to a TSV *output_file*.
+    Uses the keys of the first record as output column names.
+    Ignores extra keys in other records.
+    If records are missing keys, they will have an empty string as the value.
+
+    Parameters
+    ----------
+    records: iterator[dict]
+        Iterator that yields dict that contains sequences
+
+    output_file: str
+        Path to the output TSV file.
+        Accepts '-' to output TSV to stdout.
+    """
+    # Use the keys of the first record as output fields
+    first_record = next(records)
```

Comment:
Good catch! This is not guarded against upstream. 
Added in 3c633b43a746a678edde8e97de3359280ffc52c3 to raise an error if we encounter empty records because it seems like something went wrong if you are trying to write _nothing_ to a file!